src/hub/HubRegistry.sol

```diff
@@ -76,18 +78,27 @@ contract HubRegistry is Auth, IHubRegistry {
     }
 
     /// @inheritdoc IHubRegistry
-    function updateCurrency(PoolId poolId, AssetId currency_) external auth {
-        require(exists(poolId), NonExistingPool(poolId));
+    function updateCurrency(PoolId poolId_, AssetId currency_) external auth {
+        require(exists(poolId_), NonExistingPool(poolId_));
         require(!currency_.isNull(), EmptyCurrency());
 
-        currency[poolId] = currency_;
+        currency[poolId_] = currency_;
 
-        emit UpdateCurrency(poolId, currency_);
+        emit UpdateCurrency(poolId_, currency_);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // PermissViewionless methods
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IHubRegistry
+    function poolId(uint16 centrifugeId, uint48 postfix) public pure returns (PoolId poolId_) {
+        poolId_ = newPoolId(centrifugeId, postfix);
```

Comment:
```suggestion
    function poolId(uint48 postfix) public pure returns (PoolId poolId_) {
        poolId_ = newPoolId(centrifugeId, postfix);
        require(currency[poolId_].isNull(), PoolAlreadyRegistered())
```

Reply:
I disagree, think this is a useful lookup without the require. UI can easily check it still

Reply:
Sounds good.

Reply:
Just think the `centrifugeId` should not be part of the parameter. Or why do you think it should be?